### ISSUE TITLE
Aggregated Umbraco Property Values

### ISSUE THREAD

**Issue by leekelleher** [leekelleher/umbraco-ditto#90]
On one of my websites, I'd like to aggregate the values from a recursive Umbraco property, so rather than use recursive as a fall-back value, I can bundle all the values together.

An example (that I'm currently using) is that of additional HTML meta-tags, e.g. each page/subpage has it's own unique value which would need to be displayed/rendered.

The POCO's property would be like this...

```csharp
public IEnumerable<HtmlString> HtmlMetaTags { get; set; }
```

Currently I am hooking into the post conversion code to manually populate the property...

```csharp
foo.HtmlMetaTags = content.AncestorsOrSelf()
	.Reverse()
	.Where(x => x.HasValue("htmlMetaTags"))
	.Select(x => x.GetPropertyValue<HtmlString>("htmlMetaTags"));
```

I'm wondering if there is a way to wire this up using a custom ValueResolver (and/or inheriting from the `UmbracoPropertyAttribute`)?

Then discuss whether this could be a potential addition to the Ditto core.

**Comment by mattbrailsford** [comment]
I guess this ties in somewhat to PR #86 

I think ValueResolvers are probably the "right" solution, though I do get the feeling we would end up with tonnes of specialized ones sitting around if we always go this route. My thought was to keep using the post convert callbacks, but try and make them a bit simpler to use. I think people are always going to have custom conversions they need to run, so it's just deciding the best place for it.

In my head, I'm thinking if it's repeatable, and it's easy to represent as a value resolver, then this is the right place, but if it's a one off, or it's hard to make generic for a value resovler attribute, then it might be best to use the post convert methods proposed in PR #86

**Comment by mattbrailsford** [comment]
We happy this is resolved by PE #86? And the simplification of the ValueResolvers?

**Comment by leekelleher** [comment]
Yeah, I'll close this one off.  With my project, I already have a solution, so it's nothing urgent.

I'll explore the aggregation techniques over time - then if it looks like a good generic solution for Ditto, I'll bring it back here for discussion.
